app/util.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const request = require('request-promise');
+
+// returns a Date representing tomorrow at 12pm so that
+// we can issue darksky 'time machine' requests to get an
+// idea of the weather tomorrow at a time that should be
+// relevant to someone interested in the weather
+function getTomorrowAtNoon() {
+    let tomorrow = new Date();
+    tomorrow.setDate(tomorrow.getDate() + 1);
+    tomorrow.setHours(12);
+    return tomorrow;
+}
+
+// This url can be used to grab a frame from a DDOT (DC) traffic cam!
+// This can be used to give a rough visual of the weather. The camera is
+// at the intersection of 14th St and Independence St NW. Looks like the current
+// time in milliseconds since 1970 needs to be passed in to get the latest images.
+// The RTM client doesn't allow sending attachments (see https://api.slack.com/rtm),
+// but the Slack URL unfurling functionality automatically includes the traffic cam frame!
+// See http://app.ddot.dc.gov/ for more info.
+function getTrafficCamImageUrl() {
+    const trafficlandApiUrl = 'http://api.trafficland.com/v1.5/json/video_feeds?system=ddot&key=1594c8892d7fbd18181a8a6a44958b28&region=WAS';
+    let promise = request({ uri: trafficlandApiUrl, json: true }).then((json) => {
+        if (Array.isArray(json)) {
+            let cameraObj = json.find((element) => {
+                return element.publicId === '200146'
+            });
+
+            if (cameraObj) {
+                return cameraObj;
+            } else {
+                throw 'Unable to find desired traffic camera!';
```

Comment:
Well, since this is inside a promise, I think I need to `throw` in order to ensure  the promise is rejected and the error handler is called.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#Return_value